packages/url_launcher/ios/Classes/UrlLauncherPlugin.m

```diff
@@ -33,19 +34,20 @@ - (instancetype)initWithUrl:url withFlutterResult:result {
 
 - (void)safariViewController:(SFSafariViewController *)controller
       didCompleteInitialLoad:(BOOL)didLoadSuccessfully API_AVAILABLE(ios(9.0)) {
-  if (didLoadSuccessfully) {
-    self.flutterResult(nil);
-  } else {
-    self.flutterResult([FlutterError
-        errorWithCode:@"Error"
-              message:[NSString stringWithFormat:@"Error while launching %@", self.url]
-              details:nil]);
-  }
+  self.didLoadSuccessfully = didLoadSuccessfully;
 }
 
 - (void)safariViewControllerDidFinish:(SFSafariViewController *)controller API_AVAILABLE(ios(9.0)) {
   [controller dismissViewControllerAnimated:YES completion:nil];
   self.didFinish();
+  if (self.didLoadSuccessfully) {
```

Comment:
If the user exited the `Webview` before `safariViewController` was called, this would be `null` and it would throw a `FlutterError`. I think you could add another `if else` block that throws an error if `self.didLoadSuccessfully` is `null`. The message could say something along the lines of *"User closed Webview before url was launched"*.